objcgen/objcgenerator.cs

```diff
@@ -993,9 +989,8 @@ void GenerateDefaultValuesWrapper (ProcessedMemberBase member)
 
 			MethodBase mb = method != null ? (MethodBase)method.Method : ctor.Constructor;
 			MethodInfo mi = mb as MethodInfo;
-			string objcsig;
-			string monosig;
-			var parameters = method != null ? method.Method.GetParameters () : ctor.Constructor.GetParameters ();
+			int firstDefaultParameter = method != null ? method.FirstDefaultParameter : ctor.FirstDefaultParameter;
```

Comment:
`FirstDefaultParameter` was moved to a shared base type, can't we take advantage of this ?